library/core/resource_loader/src/main/java/org/quiltmc/qsl/resource/loader/impl/StaticResourceManagerWrapper.java

```diff
@@ -0,0 +1,28 @@
+package org.quiltmc.qsl.resource.loader.impl;
+
+import net.minecraft.resource.MultiPackResourceManager;
+import org.jetbrains.annotations.ApiStatus;
+
+/**
+ * Wrapper/duck interface for {@link MultiPackResourceManager} instances.
+ * @see StaticResourceManager
+ */
+@ApiStatus.Internal
+public interface StaticResourceManagerWrapper {
```

Comment:
This is useless, a simple `instanceof StaticResourceManager` is already sufficient for checking.

Reply:
![Screen Shot 2023-07-15 at 8 31 23 PM](https://github.com/QuiltMC/quilt-standard-libraries/assets/24525012/b88d8e13-f786-43a4-af9e-a21f1a71b9c7)

Would the mixin need to `extend ResoruceManager` for that to work? I'm presuming yes, and am going to test it while working on the upcoming commit—if it doesn't work I'll post proof as screenshot or video.

Reply:
Couldn't get it to work with both `implements AutoCloseableResourceManager` & `implements ResourceManager`. I have exactly zero clue why this is the case—logic says it should work, but it just doesn't.

![Screen Shot 2023-07-15 at 8 37 21 PM](https://github.com/QuiltMC/quilt-standard-libraries/assets/24525012/591531ca-e0dc-4e13-b51b-ddac9c14e1fb)
![Screen Shot 2023-07-15 at 8 36 03 PM](https://github.com/QuiltMC/quilt-standard-libraries/assets/24525
![Screen Shot 2023-07-15 at 8 37 48 PM](https://github.com/QuiltMC/quilt-standard-libraries/assets/24525012/0e622eeb-0b13-4c06-a33c-45fa3c3a84dc)
012/5fddc8e9-e17b-420a-b155-535f14b7a81f)

Reply:
You need to cast to object before you instanceof. Java is smart and finds impossible conditions

Reply:
Yup, turns out it was an 'IDEA and the compiler don't understand mixins' moment and not an actual error—this has been fixed and thus said wrapper interface will be gone in next commit.